### PR TITLE
Restructure the backend directory to facilitate adding more services.

### PR DESCRIPTION
**Pull request by mdivanova**

### CODE REVIEW

File: backend/api-service/src/.gcloudignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+api_inventory.go
+api.go
+logger.go
+model_*.go
+routers.go
```
Comment (josueetcom) [inline backend/api-service/src/.gcloudignore:5]: should `main.go` be added to this list? I don't see it being checked in.

**Comment by josueetcom** [comment]
It looks like you'll need to change [these lines](https://github.com/GoogleCloudPlatform/cloud-run-anthos-reference-web-app/blob/master/.github/workflows/pull-request.yaml#L100-L105) in the PR checks to use the new directory structure

File: backend/api-service/src/.gcloudignore (hunk 1 above)
Comment (mdivanova) [reply backend/api-service/src/.gcloudignore:5]: main.go is listed in backend/api-service/.gcloudignore

It gets generated one level up from these files (no change from the current directory structure)

File: backend/api-service/.gcloudignore [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.openapi-generator/
+api/
+src/api*.go
```
Comment (josueetcom) [inline backend/api-service/.gcloudignore:3]: I suggest just leaving this `.gcloudignore` file.
```suggestion
src/api_inventory.go
src/api.go
```

File: backend/api-service/src/.gcloudignore (hunk 1 above)
Comment (josueetcom) [reply backend/api-service/src/.gcloudignore:5]: Ah you're right. In that case, can't we delete this file then?

Comment (mdivanova) [reply backend/api-service/src/.gcloudignore]: Yes - that makes sense. I've deleted the nested one. Thanks!

Comment (josueetcom) [reply backend/api-service/src/.gcloudignore]: Thanks!

File: .github/workflows/pull-request.yaml [hunk 3]
```diff
@@ -97,12 +97,12 @@ jobs:
         go-version: '1.13'
     - id: gofmt
       run: |
-        GOFMT_OUT=$(gofmt -l ./backend/src)
+        GOFMT_OUT=$(gofmt -l ./backend/api-service)
```
Comment (ssmall) [inline .github/workflows/pull-request.yaml:100]: Should this be more generic so that it covers any/all services in the `backend/` directory?

Comment (mdivanova) [reply .github/workflows/pull-request.yaml]: Yep - that makes sense. Changed.

**Comment by mdivanova** [comment]
> How thoroughly have these changes been tested? Ideally they would be (manually) tested by setting up a fresh project, running all make commands, and then running the frontend e2e tests.

I did manually set up a fresh project on cloud shell and run all the make commands. Manual testing of the site looked good.

The frontend e2e tests are flaky though - a different number fail on different runs. The failure errors are not manually reproducible on the site (e.g. Not being able to find a "Create" button on the Items page). I don't think this is related to my refactoring. Are these reliably run by others?

**Comment by ssmall** [comment]
> > How thoroughly have these changes been tested? Ideally they would be (manually) tested by setting up a fresh project, running all make commands, and then running the frontend e2e tests.
> 
> I did manually set up a fresh project on cloud shell and run all the make commands. Manual testing of the site looked good.
> 
> The frontend e2e tests are flaky though - a different number fail on different runs. The failure errors are not manually reproducible on the site (e.g. Not being able to find a "Create" button on the Items page). I don't think this is related to my refactoring. Are these reliably run by others?

@zsxking can you comment on e2e test flakiness? Is this expected or should we be concerned that these changes broke something?

**Comment by mdivanova** [comment]
> > > How thoroughly have these changes been tested? Ideally they would be (manually) tested by setting up a fresh project, running all make commands, and then running the frontend e2e tests.
> > 
> > 
> > I did manually set up a fresh project on cloud shell and run all the make commands. Manual testing of the site looked good.
> > The frontend e2e tests are flaky though - a different number fail on different runs. The failure errors are not manually reproducible on the site (e.g. Not being able to find a "Create" button on the Items page). I don't think this is related to my refactoring. Are these reliably run by others?
> 
> @zsxking can you comment on e2e test flakiness? Is this expected or should we be concerned that these changes broke something?

I see the same behavior when running e2e tests on my project from last week's bugash.

**Comment by zsxking** [comment]
The flakiness is expected because of the arbitrary waiting. As long as it's not failing on the same test over and over again, it's probably fine.

**Pull request merged**